pkg/upgrade/upgrade.go

```diff
@@ -463,19 +449,33 @@ func deleteDeployments(cli client.Client, applicationNamespace string) error {
 	// In v2, Deployment selectors use a label "app.opendatahub.io/<componentName>" which is
 	// not present in v1. Since label selectors are immutable, we need to delete the existing
 	// deployments and recreated them.
+	// because we can't proceed if a deployment is not deleted, we use exponential backoff
+	// to retry the deletion until it succeeds
+	err := wait.ExponentialBackoffWithContext(context.TODO(), wait.Backoff{
+		Duration: 1 * time.Second,
```

Comment:
should we set the init time a bit higher (e.g 5sec then the 2nd is on 10sec)?
i don think for 1sec in the 1st and 2nd till be enough for deletion all 5 deployment done in time.
from my test it took less than 10sec to get the debug printed out.